### ISSUE TITLE
Caido doesn't retrieve target server output when it doesn't closes connection

### ISSUE THREAD

**Issue by nil0x42** [caido/caido#1375]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help us know what to work on
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Bug description

While auditing a buggy web server, I noticed that some queries were failing when processed through Caido. Upon further investigation (using curl), I found that such requests "partially work": the server returns HTML content but never closes the connection.

In some cases, the server enters an unconditional loop and continuously generates HTML indefinitely. Unfortunately, Caido does not handle such scenarios properly.
Expected Behavior

Caido should handle unclosed connections more robustly by:

    Reading up to a predefined request timeout.
    Returning partial results when necessary, instead of failing and returning nothing.

This behavior almost caused me to miss identifying a significant vulnerability during my audit.

> PS: I'm new to Caido but already impressed by its potential. My swap space is very grateful compared to the heavy load of Burp Suite. 😊

### How to reproduce

It's hard to say, would need to implement a buggy webserver which 'forgets' closing connections

### Environment & setup

OS: ArchLinux
OS version: 6.12.1-arch1-1
Caido client: Caido Desktop (appimage)
Caido version: 0.44.1

**Comment by Sytten** [comment]
That's an interesting side effect we didn't consider, we do get that data and timeout at some point even if the server doesn't close the connection but I don't think we have a global timeout if the server keeps sending us data.

**Comment by nil0x42** [comment]
Yes, and even when the server stops sending data, if it "forgets" to close the connection, all retrieved data is dropped from Caido's front-end point of view. Implementing a hard limit on the maximum bytes to retrieve could also be useful for scenarios where the server continuously sends data indefinitely.
